tools/istio-iptables/pkg/builder/iptables_builder_test.go

```diff
@@ -0,0 +1,116 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package builder
+
+import (
+	"reflect"
+	"testing"
+)
+
+//TODO(abhide): Add more testcases once BuildV6Restore() are implemented
+func Test_BuildV6Restore(t *testing.T) {
+	iptables := NewIptablesBuilder()
+	expected := ""
+	actual := iptables.BuildV6Restore()
+	if expected != actual {
+		t.Errorf("Output didn't match: Got: %s, Expected: %s", actual, expected)
+	}
+}
+
+//TODO(abhide): Add more testcases once BuildV4Restore() are implemented
+func Test_BuildV4Restore(t *testing.T) {
+	iptables := NewIptablesBuilder()
+	expected := ""
+	actual := iptables.BuildV4Restore()
+	if expected != actual {
+		t.Errorf("Output didn't match: Got: %s, Expected: %s", actual, expected)
+	}
+}
+
+// Test_BuildV4 tests when only ipv4 rules get added
+func Test_BuildV4(t *testing.T) {
+	iptables := NewIptablesBuilder()
+	iptables.AppendRuleV4("chain", "table", "-f", "foo", "-b", "bar")
+	iptables.InsertRuleV4("chain", "table", 2, "-f", "foo", "-b", "bar")
+	iptables.AppendRuleV4("chain", "table", "-f", "foo", "-b", "baz")
+	if err := len(iptables.rules.rulesv6) != 0; err {
+		t.Errorf("Expected rulesV6 to be empty; but got %v", iptables.rules.rulesv6)
+	}
+	actual := iptables.BuildV4()
+	expected := []string{
+		"iptables -t table -A chain -f foo -b bar", "iptables -t table -I chain 2 -f foo -b bar",
+		"iptables -t table -A chain -f foo -b baz"}
+	if !reflect.DeepEqual(actual, expected) {
+		t.Errorf("Actual and expected output mismatch; but instead got Actual: %v ; Expected: %v", actual, expected)
+	}
+	// V6 rules should be empty and return an empty slice
+	actual = iptables.BuildV6()
+	if !reflect.DeepEqual(actual, []string{}) {
+		t.Errorf("Expected V6 rules to be empty; but instead got Actual: %v", actual)
+	}
+}
+
+// Test_BuildV6 tests when only ipv6 rules get added
+func Test_BuildV6(t *testing.T) {
+	iptables := NewIptablesBuilder()
+	iptables.AppendRuleV6("chain", "table", "-f", "foo", "-b", "bar")
+	iptables.InsertRuleV6("chain", "table", 2, "-f", "foo", "-b", "bar")
+	iptables.InsertRuleV6("chain", "table", 1, "-f", "foo", "-b", "bar")
+	if err := len(iptables.rules.rulesv4) != 0; err {
+		t.Errorf("Expected rulesV4 to be empty; but got %v", iptables.rules.rulesv4)
+	}
+	actual := iptables.BuildV6()
+	expected := []string{
+		"ip6tables -t table -A chain -f foo -b bar",
+		"ip6tables -t table -I chain 2 -f foo -b bar",
+		"ip6tables -t table -I chain 1 -f foo -b bar",
+	}
+	if !reflect.DeepEqual(actual, expected) {
+		t.Errorf("Actual and expected output mismatch; but instead got Actual: %v ; Expected: %v", actual, expected)
```

Comment:
I think using `%#v` has a more readable output than `%v` for such complex types. Up to you.

Reply:
Thanks.. hadn't used `%#v` before. Changed to `%#v`